ESQL: AssertionError on aggregators

Reproduction (May be outdated):
```bash
.\gradlew :x-pack:plugin:esql:qa:server:multi-node:javaRestTest --tests "org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT.test {feature:BASELINE}" -Dtests.seed=5EED000000000101
```

Query:
```esql
from web_logs,air* | mv_expand `type` | uri_parts tSTlJsRyGBmW = "https://www.example.com:8080/path?query=1#section" | dissect airport "%{tSTlJsRyGBmW.query}" | dissect tSTlJsRyGBmW.scheme "%{JQFWZrYdzYY}" | grok JQFWZrYdzYY "%{WORD:bFylajTeOXMO} %{WORD:tSTlJsRyGBmW.path} %{WORD:OgcVNmgDw}" | stats  region = sum(scalerank) + std_dev(scalerank), siBJUQUwisn = top(OgcVNmgDw, 5, "desc"), type = count(scalerank) + median_absolute_deviation(scalerank) + weighted_avg(scalerank, 4), GRZGCKKYYmG = sample(scalerank, 8), uri = count(tSTlJsRyGBmW.port) by airport
```

Error:
```text
GenerativeIT > test > test {feature:BASELINE} FAILED
    java.lang.AssertionError: query: from web_logs,air* | mv_expand `type` | uri_parts tSTlJsRyGBmW = "https://www.example.com:8080/path?query=1#section" | dissect airport "%{tSTlJsRyGBmW.query}" | dissect tSTlJsRyGBmW.scheme "%{JQFWZrYdzYY}" | grok JQFWZrYdzYY "%{WORD:bFylajTeOXMO} %{WORD:tSTlJsRyGBmW.path} %{WORD:OgcVNmgDw}" | stats  region = sum(scalerank) + std_dev(scalerank), siBJUQUwisn = top(OgcVNmgDw, 5, "desc"), type = count(scalerank) + median_absolute_deviation(scalerank) + weighted_avg(scalerank, 4), GRZGCKKYYmG = sample(scalerank, 8), uri = count(tSTlJsRyGBmW.port) by airport
    exception: Connection reset
        at __randomizedtesting.SeedInfo.seed([5EED000000000101:D6B93FDAAEFC6CF9]:0)
        at org.junit.Assert.fail(Assert.java:89)
        at org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest.checkException(GenerativeRestTest.java:548)
        at org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest.checkPipelineException(GenerativeRestTest.java:412)
        at org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest$1.run(GenerativeRestTest.java:290)
        at org.elasticsearch.xpack.esql.generator.EsqlQueryGenerator.generatePipeline(EsqlQueryGenerator.java:220)
        at org.elasticsearch.xpack.esql.qa.rest.generative.GenerativeRestTest.test(GenerativeRestTest.java:330)
```

Node root cause:
```text
[2026-05-26T13:37:37,048][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [test-cluster-1] fatal error in thread [elasticsearch[test-cluster-1][esql_worker][T#4]], exiting
java.lang.AssertionError
    at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.BigArrays$ByteArrayAsLongArrayWrapper.get(BigArrays.java:284)
    at org.elasticsearch.compute.aggregation.CountGroupingAggregatorFunction.evaluateFinal(CountGroupingAggregatorFunction.java:302)
    at org.elasticsearch.compute.aggregation.CountGroupingAggregatorFunction.evaluateIntermediate(CountGroupingAggregatorFunction.java:274)
    at org.elasticsearch.compute.operator.HashAggregationOperator$PreparedForEvaluation.buildPage(HashAggregationOperator.java:884)
    at org.elasticsearch.compute.operator.HashAggregationOperator$MultiPageResult.next(HashAggregationOperator.java:829)
    at org.elasticsearch.compute.operator.HashAggregationOperator$MultiPageResult.next(HashAggregationOperator.java:808)
    at org.elasticsearch.compute.operator.HashAggregationOperator.getOutput(HashAggregationOperator.java:475)
    at org.elasticsearch.compute.operator.Driver.runSingleLoopIteration(Driver.java:323)
    at org.elasticsearch.compute.operator.Driver.run(Driver.java:194)
    at org.elasticsearch.compute.operator.Driver$1.doRun(Driver.java:506)
    at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
    at org.elasticsearch.compute.operator.DriverScheduler$1.doRun(DriverScheduler.java:57)
    at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
    at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:35)
    at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:1114)
    at org.elasticsearch.server@9.5.0-SNAPSHOT/org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:27)
    at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
    at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
    at java.base/java.lang.Thread.run(Thread.java:1516)
```

Another build: https://gradle-enterprise.elastic.co/s/25r7qq6kvjfl4/tests/task/:x-pack:plugin:esql:qa:server:multi-node:javaRestTest/details/org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT/test%20%7Bfeature:BASELINE%7D?page=eyJvdXRwdXQiOnsiMCI6Mn19&top-execution=1

--- a/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/CountGroupingAggregatorFunction.java
+++ b/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/CountGroupingAggregatorFunction.java
@@ -296,13 +296,14 @@ public class CountGroupingAggregatorFunction implements GroupingAggregatorFuncti
 
     private void evaluateFinal(Block[] blocks, int offset, IntVector selectedInPage) {
         try (LongVector.Builder builder = driverContext.blockFactory().newLongVectorFixedBuilder(selectedInPage.getPositionCount())) {
-            final int upTo = Math.min(selectedInPage.getPositionCount(), (int) counts.size());
-            for (int i = 0; i < upTo; i++) {
-                int si = selectedInPage.getInt(i);
-                builder.appendLong(counts.get(si));
-            }
-            for (int i = upTo; i < selectedInPage.getPositionCount(); i++) {
-                builder.appendLong(0L);
+            final int positionCount = selectedInPage.getPositionCount();
+            for (int i = 0; i < positionCount; i++) {
+                final int si = selectedInPage.getInt(i);
+                if (si < counts.size()) {
+                    builder.appendLong(counts.get(si));
+                } else {
+                    builder.appendLong(0L);
+                }
             }
             blocks[offset] = builder.build().asBlock();
         }

